_posts/2020-02-01-rom-and-dry-showcase-part-1.markdown

```diff
@@ -234,12 +234,12 @@ To use this class (and others that we will create later on), we'll need to creat
 ```ruby
 Bix::Application.boot(:persistence) do |app|
   start do
-    register('container', ROM.container(:sql, app['db.connection']))
+    register('container', ROM.container(:sql, app['db.config'].gateways[:default].connection))
```

Comment:
You're right that this line needs changing, but I just figured out you can do this:

```
register('container', ROM.container(app['db.config']))
```

Reply:
This was my first attempt as well, it did not work for me. The db.config returns a ROM::Configuration instance, where the container expects a database connection.

Reply:
Yup, there's a bit more that needs to change. I've revised Part 1 just this morning and the relevant changes for this part are: https://github.com/radar/radar.github.io/commit/57aae3af11d7790df188adb4a419f589fd078502#diff-0de0c2c310f3cf59a6ea6fdc92b1af3aR368

Reply:
awesome! I finally got to the point where I actually understand the need for `user_repo.rb` and `create_user.rb`. The dependency injection makes it so much more readable.

Reply:
It's so great! :D

Reply:
figured out we can use aliases! Love it even more.. `Import[user_repo: 'repos.user']`

Reply:
Oh great, I didn't know about those aliases.